Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm

```diff
@@ -1750,17 +1750,21 @@ void testSuspendServiceWorkerProcessBasedOnClientProcesses(UseSeparateServiceWor
     [webView2 loadRequest:server.request()];
 
```

Comment:
We may want to call `[webView _test_waitForDidFinishNavigation];` and `[webView2 _test_waitForDidFinishNavigation];` after the loadRequest calls do make sure the pages are done loading before we proceed with the test. Doesn't seem like a great way to set that the function call is working if you have to keep on calling it to make it work :)

Based on your explanation, I think this would address the flakiness without making any other changes.